src/it/scala/io/iohk/ethereum/sync/util/SyncUtils.scala

```diff
@@ -0,0 +1,78 @@
+package io.iohk.ethereum.sync.util
+
+import java.net.{InetSocketAddress, ServerSocket}
+import java.util.concurrent.TimeoutException
+
+import akka.util.ByteString
+import io.iohk.ethereum.Mocks.MockValidatorsAlwaysSucceed
+import io.iohk.ethereum.domain.{Account, Address, Block, Receipt}
+import io.iohk.ethereum.ledger.{BlockExecutionError, BlockExecutionSuccess, InMemoryWorldStateProxy}
+import io.iohk.ethereum.nodebuilder.ShutdownHookBuilder
+import monix.eval.Task
+import io.iohk.ethereum.utils.Logger
+
+import scala.concurrent.duration.FiniteDuration
+
+object SyncUtils {
+
+  object ShutdownHookBuilder extends ShutdownHookBuilder with Logger
+  class ValidatorsExecutorAlwaysSucceed extends MockValidatorsAlwaysSucceed {
+    override def validateBlockAfterExecution(block: Block, stateRootHash: ByteString, receipts: Seq[Receipt], gasUsed: BigInt)
+    : Either[BlockExecutionError, BlockExecutionSuccess] = Right(BlockExecutionSuccess)
+  }
+
+  object ValidatorsExecutorAlwaysSucceed extends ValidatorsExecutorAlwaysSucceed
+
+  def retryUntilWithDelay[A](source: Task[A], delay: FiniteDuration, maxRetries: Int)(
```

Comment:
Ths method do retry only if Task is successfull (flatMap). If we want to retry when there is exception thrown we should do it like retryOnFailure in: https://monix.io/api/3.0/monix/eval/Task.html
Possinly we could use https://github.com/cb372/cats-retry

Reply:
Now I get it. This is a utility used only in tests. So no need for overthinking it :)